### DISCUSSION TITLE
Adaptors

### DISCUSSION THREAD

**Discussion by Dark-Vinci** [Totodore/socketioxide#312] [Q&A]
Hello everyone, when are we getting a Redis/valkey adapter, this feature is so crucial to my use case.

**Comment by Totodore** [comment]
Hello,

I agree that it is quite an important feature. However it requires quite a lot of changes to the current API and also I would like to first add support for dynamic namespaces (#212) and binary nested data (#276).

**Comment by torto** [comment]
@Totodore To implement a new adapter, just use the Adapter Trait? If so, I can do it for MongoDB and Redis

**Reply by Totodore** [reply]
Hey! Thanks for the propositon. The main issue is that we need async adapters for that #86.
Once it is done (and for the moment it requires some API design specification/decision before deciding what we are going to do), it will be very nice if you want to implement the adapter trait for `MongoDB`.

**Reply by torto** [reply]
I will keep my eyes here if it is not too much to ask; when you have it done, can you ping me here so I can receive the notification?
My app will work only after that implementation, and I can help you :dancing_men:
